x-pack/plugins/security/public/management/api_keys/api_keys_grid/api_key_flyout.tsx

```diff
@@ -50,45 +56,99 @@ export interface ApiKeyFormValues {
   metadata: string;
 }
 
-export interface CreateApiKeyFlyoutProps {
+export interface ApiKeyFlyoutProps {
   defaultValues?: ApiKeyFormValues;
-  onSuccess?: (apiKey: CreateApiKeyResponse) => void;
+  onSuccess?: (
+    createdApiKeyResponse: CreateApiKeyResponse | undefined,
+    updateApiKeyResponse: UpdateApiKeyResponse | undefined
+  ) => void;
   onCancel: FormFlyoutProps['onCancel'];
+  selectedApiKey?: ApiKey;
+  readonly?: boolean;
 }
 
 const defaultDefaultValues: ApiKeyFormValues = {
   name: '',
-  expiration: '',
   customExpiration: false,
-  customPrivileges: false,
+  expiration: '',
   includeMetadata: false,
-  role_descriptors: '{}',
   metadata: '{}',
+  customPrivileges: false,
+  role_descriptors: '{}',
 };
 
-export const CreateApiKeyFlyout: FunctionComponent<CreateApiKeyFlyoutProps> = ({
+export const ApiKeyFlyout: FunctionComponent<ApiKeyFlyoutProps> = ({
   onSuccess,
   onCancel,
   defaultValues = defaultDefaultValues,
+  selectedApiKey,
+  readonly = false,
 }) => {
+  let formTitle = 'Create API Key';
+  let inProgressButtonText = 'Creating API Key…';
+
+  if (selectedApiKey) {
+    // Collect data from the selected API key to pre-populate the form
+    const doesMetadataExist = Object.keys(selectedApiKey.metadata).length > 0;
+    const doCustomPrivilegesExist = Object.keys(selectedApiKey.role_descriptors ?? 0).length > 0;
+    const daysUntilExpiration = moment(selectedApiKey.expiration).diff(moment(), 'days', true);
+    const roundedDaysUntilExpiration =
```

Comment:
Question: Do you think this is an appropriate value to display?

Reply:
Thank you for the suggestion on this - it looks so much cleaner now with the `Status`